### PR TITLE
fix: [CDS-82779]: Use http connectivity to account for proxy configurations

### PR DESCRIPTION
**Pull request by GoelPratyush99**
## Describe your changes
Early to verify the helm repo url for Anonymous Auth type we were performing a socket validation. Now we will do a Http validation and only throw exception in case http status code is -1 or the connection throws an exception like Timeout or IO. For rest cases we pass the test this was done as proxy configurations were not compatible with socket validation.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54931)

### CODE REVIEW

**Comment by GoelPratyush99** [comment]
trigger uts

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/OciHelmValidationHandler.java [hunk 1]
```diff
@@ -64,9 +70,10 @@ public ConnectorValidationResult validate(
 
       decryptEncryptedDetails(helmValidationParams);
       if (OciHelmAuthType.ANONYMOUS.equals(ociHelmConnectorDTO.getAuth().getAuthType())) {
-        URI parsedUri = helmTaskHelperBase.getParsedURI(ociUrl);
-        if (!connectableHost(parsedUri.getHost(), parsedUri.getPort())) {
-          throw new InvalidRequestException(format("Invalid oci url  %s", ociUrl));
+        String uriParsedOciUrl = helmTaskHelperBase.getParsedURI(ociUrl).toString();
+        if (!Http.connectableHttpUrl(uriParsedOciUrl.replace(OCI_PREFIX, HTTPS_PREFIX), true)
```
Comment (TarunUba1) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/OciHelmValidationHandler.java:74]: nit: we can import Http.connectableHttpUrl

**Review by TarunUba1 — DISMISSED** [review]

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/OciHelmValidationHandler.java (hunk 1 above)
Comment (abosii-harness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/OciHelmValidationHandler.java:74]: Why we do require both HTTPS and HTTP?

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/OciHelmValidationHandler.java [hunk 2]
```diff
@@ -64,9 +70,10 @@ public ConnectorValidationResult validate(
...
+            && !Http.connectableHttpUrl(uriParsedOciUrl.replace(OCI_PREFIX, HTTP_PREFIX), true)) {
+          throw new InvalidRequestException(format("Invalid oci url %s", ociUrl));
```
Comment (abosii-harness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/OciHelmValidationHandler.java:76]: Let's change the error message, it's confusing

**Review by abosii-harness — CHANGES_REQUESTED** [review]

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/OciHelmValidationHandler.java (hunk 1 above)
Comment (GoelPratyush99) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/OciHelmValidationHandler.java:74]: They might have an insecure OCI registry https://github.com/helm/helm/issues/6324#issuecomment-1687395253

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/OciHelmValidationHandler.java (hunk 2 above)
Comment (GoelPratyush99) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/OciHelmValidationHandler.java]: Done

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/OciHelmValidationHandler.java (hunk 1 above)
Comment (GoelPratyush99) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/OciHelmValidationHandler.java]: Done

**Review by abosii-harness — APPROVED** [review]

**Comment by GoelPratyush99** [comment]
trigger uts

**Review by alebuh — APPROVED** [review]

**Comment by GoelPratyush99** [comment]
trigger review

**Comment by GoelPratyush99** [comment]
trigger messagecheck

**Pull request merged**